src/components/SearchBox/SearchBox.tsx

```diff
@@ -91,6 +91,13 @@ const SearchBox = React.forwardRef<HTMLInputElement, Props>(
       onSearch && onSearch();
     };
 
+    const onKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
+      if (e.key === "Enter") {
+        internalRef.current.blur();
+        triggerSearch();
```

Comment:
Shall we update the description for `onSearch` prop, that currently says "A function that is called when the user clicks the search icon" ?